modules/distributor/config.go

```diff
@@ -37,6 +37,9 @@ type Config struct {
 	//  note that setting these two config values reduces tolerance to failures on rollout b/c there is always one guaranteed to be failing replica
 	ExtendWrites bool `yaml:"extend_writes"`
 
+	// Enables writes to metric-generator instances
+	MetricsGeneratorEnabled bool `yaml:"metrics_generator_enabled"`
```

Comment:
Wondering if we should consolidate this and `search_enabled` under some kind of features config?
```
features:
  search_enabled:
  metrics_generator_enabled:
```

Reply:
Sounds good to me. These flags are unlikely to change a lot after Tempo is set up, so having them in component specific config blocks isn't that important. We just have to caution with adding too much here.

I'm fine with moving `metrics_generator_enabled` to the global config in this PR but introducing the `features` section and moving `search_enabled` should probably be done in a separate PR. This move will require some migration code/docs.

Reply:
I've created a separate issue for this: https://github.com/grafana/tempo/issues/1287

Reply:
I've moved the flag. I noticed we are now passing both `searchEnabled` and `metricsGeneratorEnabled` to the distributor, introducing a struct here with all features might make sense.